feat(transformer-dts): `--isolatedDeclarations` dts transform

Dunqing opened the pull request:
The goal of this PR is to pass the test cases in https://github.com/microsoft/TypeScript/tree/main/tests/cases/transpile

Once this is polished, we may publish a npm package with napi bindings for people to experiment with.

Dunqing commented:
* **#3664** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3664?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Boshen commented on crates/oxc_transformer_dts/src/lib.rs:
```diff
@@ -0,0 +1,283 @@
+//! References:
+//! * <https://www.typescriptlang.org/tsconfig#target>
+//! * <https://babel.dev/docs/presets>
+//! * <https://github.com/microsoft/TypeScript/blob/main/src/compiler/transformer.ts>
```
Need to update these info

Boshen commented:
We'll follow up with a PR fixing codegen to get tests passing.

Dunqing committed:
feat(transformer-dts): `--isolatedDeclarations` dts transform (#3664)

The goal of this PR is to pass the test cases in https://github.com/microsoft/TypeScript/tree/main/tests/cases/transpile

Once this is polished, we may publish a npm package with napi bindings for people to experiment with.
